Stop further execution on error

When a step fails, the action still tries to run later steps. It should exit right away.
```
       [FATAL] Error: Errors found during processing.
          at run (/home/runner/work/_actions/w3c/spec-prod/v2/node_modules/respec/tools/respec2html.js:241:11)
          at processTicksAndRejections (node:internal/process/task_queues:96:5)
    Failed
  Error: Process completed with exit code 1.
  Error: w3c/spec-prod/v2/action.yml (Line: 107, Col: 24):
  Error: The template is not valid. w3c/spec-prod/v2/action.yml (Line: 107, Col: 24): Error reading JToken from JsonReader. Path '', line 0, position 0.
  Error: w3c/spec-prod/v2/action.yml (Line: 117, Col: 24):
  Error: The template is not valid. w3c/spec-prod/v2/action.yml (Line: 117, Col: 24): Error reading JToken from JsonReader. Path '', line 0, position 0.
  Error: w3c/spec-prod/v2/action.yml (Line: 127, Col: 24):
  Error: The template is not valid. w3c/spec-prod/v2/action.yml (Line: 127, Col: 24): Error reading JToken from JsonReader. Path '', line 0, position 0.
  Error: w3c/spec-prod/v2/action.yml (Line: 137, Col: 21):
  Error: The template is not valid. w3c/spec-prod/v2/action.yml (Line: 137, Col: 21): Error reading JToken from JsonReader. Path '', line 0, position 0.
  Error: w3c/spec-prod/v2/action.yml (Line: 147, Col: 24):
  Error: The template is not valid. w3c/spec-prod/v2/action.yml (Line: 147, Col: 24): Error reading JToken from JsonReader. Path '', line 0, position 0.
```

Ref: https://github.com/w3c/spec-prod/issues/116

--- a/action.yml
+++ b/action.yml
@@ -162,3 +162,8 @@ runs:
       env:
         INPUTS_DEPLOY: ${{ steps.prepare.outputs.deploy && toJson(fromJson(steps.prepare.outputs.deploy).w3c) }}
         OUTPUTS_BUILD: ${{ steps.build.outputs.w3c && toJson(fromJson(steps.build.outputs.w3c)) }}
+
+    - name: End
+      if: ${{ failure() }}
+      shell: bash
+      run: node -e 'require("@actions/core").setFailed("Failed")'

--- a/action.yml
+++ b/action.yml
@@ -63,7 +63,6 @@ runs:
         FORCE_COLOR: true
         PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: 1
     - name: Prepare
-      if: ${{ success() }}
       id: prepare
       run: |
         echo "::group::Prepare"
@@ -75,17 +74,15 @@ runs:
         INPUTS_GITHUB: ${{ toJSON(github) }}
 
     - name: Setup toolchain
-      if: ${{ success() }}
       run: |
         echo "::group::Setup toolchain"
         node --enable-source-maps ${{ github.action_path }}/src/setup.js
         echo "::endgroup::"
       shell: bash
       env:
-        INPUTS_TOOLCHAIN: ${{ fromJson(steps.prepare.outputs.build).toolchain }}
+        INPUTS_TOOLCHAIN: ${{ steps.prepare.outputs.build && fromJson(steps.prepare.outputs.build).toolchain }}
 
     - name: Generate Static HTML
-      if: ${{ success() }}
       id: build
       run: |
         echo "::group::Generate Static HTML"
@@ -93,81 +90,75 @@ runs:
         echo "::endgroup::"
       shell: bash
       env:
-        INPUTS_BUILD: ${{ toJSON(fromJson(steps.prepare.outputs.build)) }}
+        INPUTS_BUILD: ${{ steps.prepare.outputs.build && toJSON(fromJson(steps.prepare.outputs.build)) }}
 
     - name: Upload Build Artifacts
-      if: ${{ success() }}
       id: upload-build-artifacts
       uses: actions/upload-artifact@v2
       with:
         path: |-
-          ${{ fromJson(steps.build.outputs.gh).dest }}
-          ${{ fromJson(steps.build.outputs.w3c).dest }}
+          ${{ steps.build.outputs.gh && fromJson(steps.build.outputs.gh).dest }}
+          ${{ steps.build.outputs.w3c && fromJson(steps.build.outputs.w3c).dest }}
         name: spec-prod-result
         retention-days: 5
 
     - name: Validate hyperlinks
-      if: ${{ success() && fromJson(steps.prepare.outputs.validate).links }}
+      if: ${{ success() && steps.prepare.outputs.validate && fromJson(steps.prepare.outputs.validate).links }}
       run: |
         echo "::group::Validate hyperlinks"
         node --enable-source-maps ${{ github.action_path }}/src/validate-links.js
         echo "::endgroup::"
       shell: bash
       env:
-        OUTPUTS_BUILD: ${{ toJson(fromJson(steps.build.outputs.w3c)) }}
+        OUTPUTS_BUILD: ${{ steps.build.outputs.w3c && toJson(fromJson(steps.build.outputs.w3c)) }}
 
     - name: Validate markup
-      if: ${{ success() }}
       run: |
         echo "::group::Validate markup"
         node --enable-source-maps ${{ github.action_path }}/src/validate-markup.js
         echo "::endgroup::"
       shell: bash
       env:
-        INPUTS_VALIDATE_MARKUP: ${{ fromJson(steps.prepare.outputs.validate).markup }}
-        OUTPUTS_BUILD: ${{ toJson(fromJson(steps.build.outputs.w3c)) }}
+        INPUTS_VALIDATE_MARKUP: ${{ steps.prepare.outputs.validate && fromJson(steps.prepare.outputs.validate).markup }}
+        OUTPUTS_BUILD: ${{ steps.build.outputs.w3c && toJson(fromJson(steps.build.outputs.w3c)) }}
 
     - name: Validate pubrules (Specberus)
-      if: ${{ success() }}
       run: |
         echo "::group::Validate pubrules (Specberus)"
         node --enable-source-maps ${{ github.action_path }}/src/validate-pubrules.js
         echo "::endgroup::"
       shell: bash
       env:
-        INPUTS_VALIDATE_PUBRULES: ${{ fromJson(steps.prepare.outputs.validate).pubrules }}
-        OUTPUTS_BUILD: ${{ toJson(fromJson(steps.build.outputs.w3c)) }}
+        INPUTS_VALIDATE_PUBRULES: ${{ steps.prepare.outputs.validate && fromJson(steps.prepare.outputs.validate).pubrules }}
+        OUTPUTS_BUILD: ${{ steps.build.outputs.w3c && toJson(fromJson(steps.build.outputs.w3c)) }}
         INPUTS_W3C_API_KEY: ${{ inputs.W3C_API_KEY }}
 
     - name: Validate Web IDL
-      if: ${{ success() }}
       run: |
         echo "::group::Validate Web IDL"
         node --enable-source-maps ${{ github.action_path }}/src/validate-webidl.js
         echo "::endgroup::"
       shell: bash
       env:
-        INPUTS_VALIDATE_WEBIDL: ${{ fromJson(steps.prepare.outputs.validate).webidl }}
-        OUTPUTS_BUILD: ${{ toJson(fromJson(steps.build.outputs.w3c)) }}
+        INPUTS_VALIDATE_WEBIDL: ${{ steps.prepare.outputs.validate && fromJson(steps.prepare.outputs.validate).webidl }}
+        OUTPUTS_BUILD: ${{ steps.build.outputs.w3c && toJson(fromJson(steps.build.outputs.w3c)) }}
 
     - name: Deploy to GitHub pages
-      if: ${{ success() }}
       run: |
         echo "::group::Deploy to GitHub pages"
         node --enable-source-maps ${{ github.action_path }}/src/deploy-gh-pages.js
         echo "::endgroup::"
       shell: bash
       env:
-        INPUTS_DEPLOY: ${{ toJson(fromJson(steps.prepare.outputs.deploy).ghPages) }}
-        OUTPUT_DIR: ${{ fromJson(steps.build.outputs.gh).root }}
+        INPUTS_DEPLOY: ${{ steps.prepare.outputs.deploy && toJson(fromJson(steps.prepare.outputs.deploy).ghPages) }}
+        OUTPUT_DIR: ${{ steps.build.outputs.gh && fromJson(steps.build.outputs.gh).root }}
 
     - name: Deploy to W3C
-      if: ${{ success() }}
       run: |
         echo "::group::Deploy to W3C"
         node --enable-source-maps ${{ github.action_path }}/src/deploy-w3c-echidna.js
         echo "::endgroup::"
       shell: bash
       env:
-        INPUTS_DEPLOY: ${{ toJson(fromJson(steps.prepare.outputs.deploy).w3c) }}
-        OUTPUTS_BUILD: ${{ toJson(fromJson(steps.build.outputs.w3c)) }}
+        INPUTS_DEPLOY: ${{ steps.prepare.outputs.deploy && toJson(fromJson(steps.prepare.outputs.deploy).w3c) }}
+        OUTPUTS_BUILD: ${{ steps.build.outputs.w3c && toJson(fromJson(steps.build.outputs.w3c)) }}

--- a/action.yml
+++ b/action.yml
@@ -166,5 +166,7 @@ runs:
     - name: End
       if: ${{ failure() }}
       shell: bash
-      run: node -e 'require("@actions/core").setFailed("Failed. See details above.")'
+      run: |
+        # Failed
+        node -e 'require("@actions/core").setFailed("Failed. See details above.")'
       working-directory: ${{ github.action_path }}

--- a/action.yml
+++ b/action.yml
@@ -63,6 +63,7 @@ runs:
         FORCE_COLOR: true
         PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: 1
     - name: Prepare
+      if: ${{ success() }}
       id: prepare
       run: |
         echo "::group::Prepare"
@@ -74,6 +75,7 @@ runs:
         INPUTS_GITHUB: ${{ toJSON(github) }}
 
     - name: Setup toolchain
+      if: ${{ success() }}
       run: |
         echo "::group::Setup toolchain"
         node --enable-source-maps ${{ github.action_path }}/src/setup.js
@@ -83,6 +85,7 @@ runs:
         INPUTS_TOOLCHAIN: ${{ fromJson(steps.prepare.outputs.build).toolchain }}
 
     - name: Generate Static HTML
+      if: ${{ success() }}
       id: build
       run: |
         echo "::group::Generate Static HTML"
@@ -93,6 +96,7 @@ runs:
         INPUTS_BUILD: ${{ toJSON(fromJson(steps.prepare.outputs.build)) }}
 
     - name: Upload Build Artifacts
+      if: ${{ success() }}
       id: upload-build-artifacts
       uses: actions/upload-artifact@v2
       with:
@@ -103,7 +107,7 @@ runs:
         retention-days: 5
 
     - name: Validate hyperlinks
-      if: ${{ fromJson(steps.prepare.outputs.validate).links }}
+      if: ${{ success() && fromJson(steps.prepare.outputs.validate).links }}
       run: |
         echo "::group::Validate hyperlinks"
         node --enable-source-maps ${{ github.action_path }}/src/validate-links.js
@@ -113,6 +117,7 @@ runs:
         OUTPUTS_BUILD: ${{ toJson(fromJson(steps.build.outputs.w3c)) }}
 
     - name: Validate markup
+      if: ${{ success() }}
       run: |
         echo "::group::Validate markup"
         node --enable-source-maps ${{ github.action_path }}/src/validate-markup.js
@@ -123,6 +128,7 @@ runs:
         OUTPUTS_BUILD: ${{ toJson(fromJson(steps.build.outputs.w3c)) }}
 
     - name: Validate pubrules (Specberus)
+      if: ${{ success() }}
       run: |
         echo "::group::Validate pubrules (Specberus)"
         node --enable-source-maps ${{ github.action_path }}/src/validate-pubrules.js
@@ -134,6 +140,7 @@ runs:
         INPUTS_W3C_API_KEY: ${{ inputs.W3C_API_KEY }}
 
     - name: Validate Web IDL
+      if: ${{ success() }}
       run: |
         echo "::group::Validate Web IDL"
         node --enable-source-maps ${{ github.action_path }}/src/validate-webidl.js
@@ -144,6 +151,7 @@ runs:
         OUTPUTS_BUILD: ${{ toJson(fromJson(steps.build.outputs.w3c)) }}
 
     - name: Deploy to GitHub pages
+      if: ${{ success() }}
       run: |
         echo "::group::Deploy to GitHub pages"
         node --enable-source-maps ${{ github.action_path }}/src/deploy-gh-pages.js
@@ -154,6 +162,7 @@ runs:
         OUTPUT_DIR: ${{ fromJson(steps.build.outputs.gh).root }}
 
     - name: Deploy to W3C
+      if: ${{ success() }}
       run: |
         echo "::group::Deploy to W3C"
         node --enable-source-maps ${{ github.action_path }}/src/deploy-w3c-echidna.js

--- a/action.yml
+++ b/action.yml
@@ -166,4 +166,5 @@ runs:
     - name: End
       if: ${{ failure() }}
       shell: bash
-      run: node -e 'require("@actions/core").setFailed("Failed")'
+      run: node -e 'require("@actions/core").setFailed("Failed. See details above.")'
+      working-directory: ${{ github.action_path }}

